### ISSUE TITLE
Federated sharing: volatile edit permission

### ISSUE THREAD

**Issue by michaelstingl** [owncloud/core#24032]
### Steps to reproduce
1. Create federated share with no edit permission
2. Then grant edit permission later

### Expected behaviour
Edit permission should be permanent

### Actual behaviour
Edit permission don't get saved permanent. (after switching to Favorites section or reloading the page)

![federated-sharing-edit-permission](https://cloud.githubusercontent.com/assets/214010/14565230/5a2afe76-032a-11e6-8629-4713bfc61d47.gif)

### Server configuration
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
PHP 5.5.9

**ownCloud version:** (see ownCloud admin page)
9.0.1

**Updated from an older ownCloud or fresh install:**
fresh install

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
# sudo -u www-data php occ app:list
Enabled:
  - activity: 2.2.1
  - comments: 0.2
  - dav: 0.1.6
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - templateeditor: 0.1
  - updatenotification: 0.1.0
Disabled:
  - encryption
  - external
  - files_external
  - user_external
  - user_ldap
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

### Client configuration
**Browser:**
Chrome 49

**Operating system:**
Mac OS X 10.11.4

### Logs

no messages appear in the logs (owncloud.log and apache log).

@MorrisJobke 

00005231

**Comment by michaelstingl** [comment]
An user reported this for 8.2.3 too:

> When you share a folder with someone, the default permission is to allow the receiver to write to the share. This seems to be represented by a 7 in the column "permissions" in oc_share. When the sharer changes this, the 7 changes to a 1. But when he tries to give back the write permissions and clicks on the box in the web interface (I think it is "can edit"), the permission in the database never changes back to 7. We saw this with owncloud 8.2.3

**Comment by MorrisJobke** [comment]
sounds like a JS bug - cc @rullzer @PVince81

**Comment by rullzer** [comment]
I think this will be fixed with https://github.com/owncloud/core/pull/23918

**Comment by rullzer** [comment]
Aah but that of course won't help in 9.x... I'll see if there is a quick fix for this.

**Comment by PVince81** [comment]
Setting to 9.0.2 since we have a PR ready to merge: https://github.com/owncloud/core/pull/24108

**Comment by rullzer** [comment]
Merged. Closing this

**Comment by MorrisJobke** [comment]
@michaelstingl This will also be fixed in 8.2.4 ;)
